### DISCUSSION TITLE
Treatlife Fan Control (no light)

### DISCUSSION THREAD

**Discussion by JJGira** [milo526/homebridge-tuya-web#251] [General]
Hoping you can help.  I just added a [Treatlife Fan Control](https://www.amazon.com/gp/product/B08P5D3ZKW/ref=ppx_yo_dt_b_asin_title_o07_s00?ie=UTF8&psc=1) to my SmartLife app.   
I already have SmartLife added to my Homebridge (great plugin, btw!).   
For some reason, the new fan doesn't show up though.  It does show up in Alexa (using the SmartLife skill) so I know it is showing up in SmartLife (aside from the fact it is in the app, lol).  Any ideas why?  When I restart Homebridge and it is loading the TuyaWebPlatform section, I do see 
> [@milo526/homebridge-tuya-web] This plugin generated a warning from the characteristic 'Brightness': characteristic was supplied illegal value: number 392 exceeded maximum of 100. See https://git.io/JtMGR for more info.

Not sure if this is from this new device, but it is the only device I am missing.  

Thanks in advance!

**Comment by JJGira** [comment]
Also, I noticed this in the log:
[07/04/2021, 15:53:53] [TuyaWebPlatform] Missing required `id` property on device configuration, received:
{
  "device_type": "form"
}

**Reply by milo526** [reply]
Please post the result of [the steps under "How to check whether the api this library uses can control your device".](https://github.com/milo526/homebridge-tuya-web#how-to-check-whether-the-api-this-library-uses-can-control-your-device)

Without this it is hard for me to help you with regards to this specific issue.

Your configuration is wrong (this is due to a discrepancy between hoops and this plugin) but it should not affect the working of the plugin (except for the warning that was shown (Missing required id property on device configuration).

The brightness related warning is also nothing to worry about and I'm working on a fix.

**Reply by JJGira** [reply]
Thank you for the reply.  I ran the script and the fan control wasn't listed in the output.  To get around it, I created scenes (Tap-to-Run) in the Smart Life app and imported the scenes.  Since these are the only scenes I have in SL, it works out.  

Thank you for your hard work on this plug in!  Please let me know if there is something I can do to help get this control in your plug in (if that is possible or something you want to do).  Have a great day!
